tflearn/callbacks.py

```diff
@@ -9,51 +9,78 @@
 
 class Callback(object):
     """ Callback base class. """
-
-    def __init__(self, **kwargs):
+    def on_train_begin(self, training_state):
         pass
 
-    def on_epoch_begin(self, **kwargs):
+    def on_epoch_begin(self, training_state):
         pass
 
-    def on_epoch_end(self, **kwargs):
+    def on_batch_begin(self, training_state):
         pass
 
-    def on_sub_epoch_begin(self, **kwargs):
+    def on_sub_batch_begin(self, training_state):
         pass
 
-    def on_sub_epoch_end(self, **kwargs):
```

Comment:
@aymericdamien I have removed `on_sub_epoch_end` and `on_sub_epoch_begin` which are useless to me. It's good ?